File: python-backend/services/comprehend.py
import boto3
from botocore.exceptions import ClientError

# Configure AWS SDK
aws_config = {
    'aws_access_key_id': '###',
    'aws_secret_access_key': '###',
    'region_name': '###'
}

# Initialize the AWS Comprehend client
comprehend_client = boto3.client('comprehend', **aws_config)

# Function to analyze sentiment using AWS Comprehend
def analyzeSentiment(text):
    params = {
        'TextList': [text],
        'LanguageCode': 'en'  # Specify the language code; change if needed
    }

    try:
        response = comprehend_client.batch_detect_sentiment(**params)
        sentiment_data = response['ResultList'][0]
        sentiment = sentiment_data['Sentiment']
        return sentiment  # Returns the sentiment analysis results
    except ClientError as error:
        raise Exception(f"Comprehend Error: {error.response['Error']['Message']}")

# Example usage:
# sentiment = analyze_sentiment("I am happy today!")
# print(sentiment)

File: backend/services/comprehend.js
const AWS = require('aws-sdk');
AWS.config.update({
  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
  region: process.env.AWS_REGION
});

const comprehend = new AWS.Comprehend();

const analyzeSentiment = async (text) => {
  const params = {
    TextList: [text],
    LanguageCode: 'en', // Specify the language code; change if needed
  };

  try {
    const data = await comprehend.batchDetectSentiment(params).promise();
    const sentimentData = data.ResultList[0];
    console.log(sentimentData);
    const sentiment = sentimentData.Sentiment;
    return sentiment; 
  } catch (error) {
    throw new Error(`Comprehend Error: ${error.message}`);
  }
};

module.exports = { analyzeSentiment };
